modules/local/align_bwa/main.nf

```diff
@@ -25,52 +25,101 @@ process INDEX_FASTA {
     """
 }
 
-process ALIGNMENT_BAM {
+process FEATURED_ALIGNMENT {
 
     /*
     This module aligns reads to the reference using specified arguments and produces output in the form of FASTQ.GZ, BAM and BAM.BAI files.
+    It also runs:
+     - depth generation using jgi_summarize_bam_contig_depths if get_depth=true
+     - concoct coverage table if get_concoct_table=true
     */
 
     label 'process_medium'
 
     tag "${meta.id} align to ${ref_fasta}"
 
-    container 'quay.io/microbiome-informatics/bwamem2:2.2.1'
+    container 'quay.io/microbiome-informatics/bwa_metabat_concoct:2.2.1_2.16_1.1.0'
 
     input:
     tuple val(meta), path(reads), path(ref_fasta), path(ref_fasta_index)
+    val(get_depth)
+    val(get_concoct_table)
+    val(bed)
 
     output:
-    tuple val(meta), path(ref_fasta), path("output/${meta.id}_sorted.bam"), path("output/${meta.id}_sorted.bam.bai"), emit: bam
+    // tuple val(meta), path(ref_fasta), path("output/${meta.id}_sorted.bam"), path("output/${meta.id}_sorted.bam.bai"), emit: bam
     path "versions.yml"                                                                                             , emit: versions
+    tuple val(meta), path("*.txt.gz"), emit: depth
+    tuple val(meta), path("*.tsv"), path("concoct*.fasta"), emit: concoct_data
 
     script:
-    def input_reads = "";
-    if ( meta.single_end ) {
-        input_reads = "${reads[0]}";
-    } else {
-        input_reads = "${reads[0]} ${reads[1]}"
-    }
+
+    def prefix = task.ext.prefix ?: "${meta.id}"
 
     def samtools_args = task.ext.alignment_args
 
+    def jgi_summarize_bam_contig_depths_args = task.ext.jgi_summarize_bam_contig_depths_args ?: ''
```

Comment:
I don't see this `task.ext.jgi_summarize_bam_contig_depths_args` anywhere in modules.config, is it anywhere or am I just dumb 😬

Reply:
I added it just in case.. to show that arguments can be adjustable (taken from nf-core). It should be empty in config. I don't know does it make sense to add it to config and set to empty.